### PR TITLE
Auth and Maps API added

### PR DESCRIPTION
**Pull request by thefredo1000**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] PR has a descriptive title
- [ ] PR contains closely related changes only (should not have many unrelated changes)
- [ ] STEP pod members have been added as reviewers
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### **Before the Authentication was added**
Screenshot 2020-07-09 at 11 10 04 AM - Display 1
Any one could comment without auth. 😔

#### Before logging in
Screenshot 2020-07-09 at 11 11 21 AM - Display 1
To comment, the user needs to be authenticated. 😲
#### After logging in
Screenshot 2020-07-09 at 11 10 41 AM - Display 1
When the user is authenticated, the option to comment its now available. 😄

#### **Before the Maps API was added**
Screenshot 2020-07-09 at 11 21 42 AM - Display 1
No Google Maps visible. 😲

#### **After the Maps API was added**
Screenshot 2020-07-09 at 11 23 31 AM - Display 2
Google Maps visible, it even shows my home town. 😎

### CODE REVIEW

File: ayup/portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      // TODO: Find a more efficient way to put this on the html
```
Comment (brettapeters) [inline ayup/portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:38]: You could use a template engine like JSP.

https://happycoding.io/tutorials/java-server/jsp

File: ayup/portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -120,3 +121,36 @@ function createDeleteButton() {
...
+  a.parentNode.replaceChild(logInFormContainer, a);
+}
+
+/** Creates the google maps div */
+(function(exports) {
+  "use strict";
+
+  function initMap() {
+    exports.map = new google.maps.Map(document.getElementById("map"), {
+      center: {
```
Comment (brettapeters) [inline ayup/portfolio/src/main/webapp/script.js:147]: Consider storing these coordinates in a local variable with a descriptive name

```js
const torreonLatLng = {
  lat: 25.5487519,
  lng: -103.4719567
};
```

File: ayup/portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java (hunk 1 above)
Comment (thefredo1000) [reply ayup/portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:38]: I'll look into it, thank you!

**Pull request merged**